README.md

```diff
@@ -98,4 +100,41 @@ We can extend the concept of the `as` syntax by generalizing the right hand side
+## FAQ
+
+### Why not out of band?
+
+Why not both? The champions of this proposal think that exploring both an in- and out of band solutions to this problem are desirable. Depending on the environment in which JavaScript is being executed in either approach could be seen as desirable.
+
+While an in band solution is more verbose, it is also more straightforward for developers to adopt. For large project with many dependencies developers will not have to worry about creating a large manifest by compiling the meta data of all of their dependencies. Module authors will also not have to worry about shipping manifest in order for consumers to be able to run their modules.
+
+The [import-maps proposal][] is a great example of an out of band manifest that could be created at installation time, but it's focus is on modules and at this time all necessary information to generate an applications import map could be found in the package.json of it's dependencies.
+
+While there very well could be a future where an out of band manifest could be used for the purpose of identifying module type, the amount of infrastructure necessary for ecosystem adoption is significantly higher than the in band solution proposed here.
+
+### Are there cross-environment concerns?
+
+Absolutely. This proposal is not attempting to standardize what `types` hosts will be allowed to support. It will be up to each runtime to decide which `types` to support, as well as how to handle the the case of an unsupported module `type` being imported.
+
+Module specifiers are an example of prior art for this type of decision making. It is left up to the host how to resolve a module specifier into a URL that will eventually be loaded. The [import-maps proposal][] is a great example of work done outside of TC39 to allow hosts to share behavior for undefined parts of the spec.
+
+Independent of the potential cross-environment concerns it is not a problem that is specific to an in band solution. An out of band solution would also suffer from the risk of inconsistent implementation or support across runtimes.
+
+### Why not pursue the generalized constant form today?
+
+This proposal was initially created to serve a very specific purpose, unblocking security concerns raised about importing alternative module types (JSON, CSS, HTML). While there is no debate that the generalized constant form would be useful, it does pose the risk of increasing cross-environment concerns mentioned above.
+
+The champion group is very much in support of pursuing the generalized constant form, but think that it would be prudent to focus the proposal to a single string initially.
+
+### How would this proposal work with caching?
+
+At the moment there is nothing in the specification text that refers to caching. While a caching strategy would likely be dependent on the type of metadata included, for the generalized constant form, this proposal would defer to the policy of environments to decide how and what to cache.
```

Comment:
This isn't true. See https://tc39.es/ecma262/#sec-hostresolveimportedmodule, the third bullet point. The specification requires that, if you pass up the same module specifier and referrer, you get the same module. In practice, there's *additional* caching that's shared between multiple referrers.

After thinking about this some more, I think we should use a similar guarantee when module attributes are included: If you pass the same specifier, referrer *and module attributes*, you get the same module. But hosts will probably do additional caching and coalescing beyond that. (E.g., in HTML, with types, it would always give you the same module based on the same specifier and referrer, or an error.)

Reply:
Took a stab at text PTAL